src/compiler/checker.ts

```diff
@@ -13293,6 +13299,11 @@ namespace ts {
                         type = checkExpressionForMutableLocation((<ShorthandPropertyAssignment>memberDecl).name, checkMode);
                     }
 
+                    if (jsdocType) {

```

Comment:
you probably need to flow this as a contextual type. talk to @rbuckton about this, it may be easier that he do that bit as part of  https://github.com/Microsoft/TypeScript/issues/15618#issuecomment-300632637